sktime/forecasting/rbf_forecaster.py

```diff
@@ -0,0 +1,575 @@
+        """Check if torch is available and raise appropriate error if not."""
+        if not _check_soft_dependencies("torch", severity="none"):
+            raise ModuleNotFoundError(
+                "torch is a required optional dependency for RBFForecaster. "
+                "Please install torch to use this forecaster. "
+                "Instructions can be found at https://pytorch.org/get-started/locally/"
+            )
+        return True
+
+    def build_network(self, input_size):
+        """Build the RBF network architecture."""
+        output_size = 1
+
+        self.model = RBFNetwork(
+            input_size=input_size,
+            hidden_size=self.hidden_size,
+            output_size=output_size,
+            centers=self.centers,
+            gamma=self.gamma,
+            rbf_type=self.rbf_type,
+            linear_layers=self.linear_layers,
+        ).to(self.device)
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit the RBF-based model to the provided time series data.
+
+        Parameters
+        ----------
+        y : pd.Series
+            The time series data to be fitted.
+        X : optional
+            Additional exogenous data (not used).
+        fh : optional
+            Forecasting horizon (not used).
+        """
+        import torch
+        from torch.utils.data import DataLoader, TensorDataset
+
+        self._y = y.copy()
+        if isinstance(y, pd.Series):
+            y_scaled = self.scaler.fit_transform(y.values.reshape(-1, 1))
+        else:
+            y_scaled = self.scaler.fit_transform(y.values)
+
+        X_train, y_train = self._create_windows(y_scaled)
+        X_tensor = torch.FloatTensor(X_train).to(self.device)
+        y_tensor = torch.FloatTensor(y_train).to(self.device)
+
+        dataset = TensorDataset(X_tensor, y_tensor)
+        dataloader = DataLoader(dataset, batch_size=self.batch_size, shuffle=True)
+
+        input_size = self.window_length
+        self.build_network(input_size)
+
+        criterion = self._instantiate_criterion()
+        optimizer = None
+
+        self.model.train()
+        for epoch in range(self.epochs):
+            total_loss = 0
+            for batch_X, batch_y in dataloader:
+                if optimizer is None:
+                    _ = self.model(batch_X)
+                    optimizer = self._instantiate_optimizer()
+
+                optimizer.zero_grad()
+                outputs = self.model(batch_X)
+                loss = criterion(outputs, batch_y)
+                loss.backward()
+                optimizer.step()
+                total_loss += loss.item()
+
+        return self
+
+    def _predict(self, fh, X=None):
+        """Generate predictions for the specified forecasting horizon.
+
+        Parameters
+        ----------
+        fh : ForecastingHorizon
+            The forecasting horizon for the predictions.
+        X : optional
+            Additional exogenous data (not used).
+        """
+        import torch
```

Comment:
I think this import is unnecessary. Since if the interpreter reaches this line, torch is already imported or?

Reply:
yeah sorry I missed it, added while I was creating and testing the class and might have missed it (as much as I can remember)